chronolog.py

```diff
@@ -0,0 +1,88 @@
+import os.path
+import regex as re
+
+import ChronologBackend.nlp as nlp
+import argparse
+from argparse import RawTextHelpFormatter
+
+DEFAULT_SPLITS = 10
+
+NUMERAL_PATTERN = re.compile(
+    r"""M{0,3}(?:CM|CD|D?C{0,3})?(?:XC|XL|L?X{0,3})?(?:IX|IV|V?I{0,3})?\n\n$""",
+    re.IGNORECASE)
+CHAPTER_NUMERAL_PATTERN = re.compile(
+    r"""(?:Chapter M{0,3}(?:CM|CD|D?C{0,3})?(?:XC|XL|L?X{0,3})?(?:IX|IV|V?I{0,3})?)""",
+    re.IGNORECASE)
+DIGIT_PATTERN = re.compile(r"""[0-9]+\n\n""", re.VERBOSE)
+CHAPTER_DIGIT_PATTERN = re.compile(r"""Chapter [0-9]+""", re.IGNORECASE)
+
+PATTERNS_DICT = {
+    "numeral": NUMERAL_PATTERN,
+    "chapter_digit": CHAPTER_DIGIT_PATTERN,
+    "chapter_numeral": CHAPTER_NUMERAL_PATTERN,
+    "digit": DIGIT_PATTERN}
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        formatter_class=RawTextHelpFormatter,
+        prog='Chronolog',
+        description="""Chronolog Book Processor Version 0.0.1"""
+    )
+    parser.add_argument('filename', type=str)
+    parser.add_argument(
+        '--chapterRegex', '-c',
+        required=False,
+        type=str,
+        metavar="CR",
+        help="""Regex Options:
+                - chapter_numeral   = Chapter + numeral
+                - chapter_digit     = Chapter + digit
+                - digit             = Digit only
+                - numeral           = Numeral only
+                You may provide your own alternative chapter regex in the form of a regex string.
+                Please make sure your regex is well-formed to avoid undesirable results.
+                Note that in any user-defined regex, there must be NO capturing groups.
+                """
+    )
+    parser.add_argument('--quiet', '-q', required=False, action='store_true',
+                        help="Suppresses updates on progress of model.")
+    parser.add_argument(
```

Comment:
What happens if there are spaces in the novel's title? Are the spaces replaced with underscores?